GTBitsOfGood/southface
# sh/183-allowNewBuildingTypes

## Allow Card Titles to be Edited

Issue Number(s): #183 

What does this PR change and why?

Changes both the frontend and backend to support new building types added by admins. 
New changes on the backend include:

- Schema for building types
- BuildingTypes collection in database reflective of existing 3 building types

It is important to note that there were no changes made to the card schema. Since a card can be associated with multiple buildingTypes, a many-to-many relationship exists between cards and buildingTypes. The assumption this feature is implemented with is that unique indexing by name can be done for buildingTypes. The only difference between using buildingType names instead of their objectids, is that it makes the procedure for editing existing buildingType names a slightly more complex operation. So that is important to keep in mind when that feature is implemented. 

New changes on the frontend include:

- Button on library page to create new building types for admins
- Modal and flow to create new building types

Changes to get existing frontend integrated:
- Library now fetches from buildingTypes collection instead of constants
- Adding new standard fetches from buildingTypes collection instead of constants

### Checklist

- [x] API/Backend for adding a new building type works.
- [x] Database has been adjusted for entirely new building types to be added.
- [x] A new building type can contain its own set of cards.
- [x] Only admin can edit building types

### How to Test

Login as an admin. On the main page, click the new create building type button and try the different flows. After adding a new building type, try adding a new standard associated with type and see if it displays correctly.
